src/shim.c

```diff
@@ -360,25 +360,23 @@ static CHECK_STATUS check_db_cert_in_ram(EFI_SIGNATURE_LIST *CertList,
 			CertSize = CertList->SignatureSize - sizeof(EFI_GUID);
 			dprint(L"trying to verify cert %d (%s)\n", i++, dbname);
 			if (verify_x509(Cert->SignatureData, CertSize)) {
-				if (verify_eku(Cert->SignatureData, CertSize)) {
```

Comment:
I prefer the EKU check being here versus pushed down and hidden inside the crypto library, business logic should be at this level, not hidden in the crypto library IMHO...

Reply:
Perhaps verify_x509() could have an EKU parameter?  Defer to @vathpela and @mjg59